server/rest/src/main/java/org/infinispan/rest/resources/ServerHealthCheckResource.java

```diff
@@ -0,0 +1,74 @@
+package org.infinispan.rest.resources;
+
+import static org.infinispan.rest.framework.Method.GET;
+import static org.infinispan.rest.framework.Method.HEAD;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.commons.CacheException;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.factories.GlobalComponentRegistry;
+import org.infinispan.lifecycle.ComponentStatus;
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.rest.InvocationHelper;
+import org.infinispan.rest.NettyRestResponse;
+import org.infinispan.rest.framework.ResourceHandler;
+import org.infinispan.rest.framework.RestRequest;
+import org.infinispan.rest.framework.RestResponse;
+import org.infinispan.rest.framework.impl.Invocations;
+import org.infinispan.security.actions.SecurityActions;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+public class ServerHealthCheckResource implements ResourceHandler {
+
+   private final InvocationHelper helper;
+
+   public ServerHealthCheckResource(InvocationHelper helper) {
+      this.helper = helper;
+   }
+
+   @Override
+   public Invocations getInvocations() {
+      return new Invocations.Builder()
+
+            .invocation().methods(GET, HEAD).path("/health/live")
+            .handleWith(this::notifyServerRunning)
+
+            .invocation().methods(GET, HEAD).path("/health/ready")
+            .handleWith(this::verifyServerReady)
+
+            .create();
+   }
+
+   public CompletionStage<RestResponse> notifyServerRunning(RestRequest request) {
+      NettyRestResponse.Builder builder = helper.newResponse(request);
+      return CompletableFuture.completedFuture(builder.status(HttpResponseStatus.OK).build());
+   }
+
+   public CompletionStage<RestResponse> verifyServerReady(RestRequest request) {
+      NettyRestResponse.Builder builder = helper.newResponse(request);
+      DefaultCacheManager dcm = helper.getServer().getCacheManager();
+      HttpResponseStatus status = HttpResponseStatus.SERVICE_UNAVAILABLE;
+      if (dcm.getStatus().allowInvocations()) {
+         if (dcm.getCacheNames().stream().anyMatch(this::isCacheReady))
```

Comment:
I don't think this is sufficient. I thought we wanted the readiness probe to only pass if all known caches were HEALTHY and not rebalancing?

Also, shouldn't we take into account the state of our internal caches? Looking at `DefaultCacheManager#getCacheNames` we appear to filter these from the returned list of names.

Reply:
Great, let me update real quick.